sdks/java/io/kinesis/src/main/java/org/apache/beam/sdk/io/kinesis/KinesisIO.java

```diff
@@ -420,6 +426,47 @@ public Read withCustomWatermarkPolicy(WatermarkPolicyFactory watermarkPolicyFact
       return toBuilder().setWatermarkPolicyFactory(watermarkPolicyFactory).build();
     }
 
+    /**
+     * Specifies the rate limit policy as BackoffRateLimiter.
+     *
+     * @param fluentBackoff The {@code FluentBackoff} used to create the backoff policy.
+     */
+    public Read withBackoffRateLimitPolicy(FluentBackoff fluentBackoff) {
+      checkArgument(fluentBackoff != null, "fluentBackoff cannot be null");
+      return toBuilder()
+          .setRateLimitPolicyFactory(RateLimitPolicyFactory.withBackoff(fluentBackoff))
+          .build();
+    }
+
+    /**
+     * Specifies the rate limit policy as FixedDelayRateLimiter with the default delay of 1 second.
+     */
+    public Read withFixedDelayRateLimitPolicy() {
+      return toBuilder().setRateLimitPolicyFactory(RateLimitPolicyFactory.withFixedDelay()).build();
+    }
+
+    /**
+     * Specifies the rate limit policy as FixedDelayRateLimiter with the given delay.
+     *
+     * @param delay Denotes the fixed delay duration.
+     */
+    public Read withFixedDelayRateLimitPolicy(Duration delay) {
```

Comment:
Please, add an example into KinesisIO javadoc how to use it.

Reply:
> 1. Have we try to use Kinesis's RetryPolicy and BackoffStrategy? This look like we reengineer this feature, which already supported by Kinesis's API. Here is the example: https://www.programcreek.com/java-api-examples/?api=com.amazonaws.retry.RetryPolicy. If we already tried it, then what would be the reasons not to use the Kinesis's one ?

I'm not sure if RetryPolicy and BackoffStrategy apply to LimitExceededException / ProvisionedThroughputExceededException but I can look into that. If so I think it makes sense to just configure this in the Kinesis client instead of having a BackoffRateLimitPolicy. What do you think @aromanenko-dev and @lukecwik ?